app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/zetime/ZeTimePreferenceActivity.java

```diff
@@ -0,0 +1,461 @@
+                return true;
+            }
+        });
+
+        final Preference dnd_start = findPreference(ZeTimeConstants.PREF_DO_NOT_DISTURB_START);
+        dnd_start.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newVal) {
+                GBApplication.deviceService().onSendConfiguration(ZeTimeConstants.PREF_DO_NOT_DISTURB);
+                return true;
+            }
+        });
+
+        final Preference dnd_end = findPreference(ZeTimeConstants.PREF_DO_NOT_DISTURB_END);
+        dnd_end.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newVal) {
+                GBApplication.deviceService().onSendConfiguration(ZeTimeConstants.PREF_DO_NOT_DISTURB);
+                return true;
+            }
+        });
+
+        final Preference caloriesType = findPreference(ZeTimeConstants.PREF_CALORIES_TYPE);
+        caloriesType.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newVal) {
+                GBApplication.deviceService().onSendConfiguration(ZeTimeConstants.PREF_CALORIES_TYPE);
+                return true;
+            }
+        });
+
+        final Preference timeFormat = findPreference(ZeTimeConstants.PREF_TIME_FORMAT);
+        timeFormat.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newVal) {
+                GBApplication.deviceService().onSendConfiguration(ZeTimeConstants.PREF_TIME_FORMAT);
+                return true;
+            }
+        });
+
+        final Preference dateFormat = findPreference(ZeTimeConstants.PREF_DATE_FORMAT);
+        dateFormat.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newVal) {
+                GBApplication.deviceService().onSendConfiguration(ZeTimeConstants.PREF_DATE_FORMAT);
+                return true;
+            }
+        });
+
+        final Preference inactivityAlert = findPreference(ZeTimeConstants.PREF_INACTIVITY_ENABLE);
+        inactivityAlert.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newVal) {
+                invokeLater(new Runnable() {
+                    @Override
+                    public void run() {
+                    GBApplication.deviceService().onSendConfiguration(ZeTimeConstants.PREF_INACTIVITY_KEY);
+                    }
```

Comment:
And since many of them send the same PREF_INACTIVITY_KEY, why not reuse one listener instance?

And for the other cases, one could probably use a method like this
```
private void addPreferenceHandlerFor(final String preferenceKey) {
	Preference pref = findPreference(preferenceKey);
	pref.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
		@Override
		public boolean onPreferenceChange(Preference preference, Object newVal) {
			invokeLater(new Runnable() {
				@Override
				public void run() {
					GBApplication.deviceService().onSendConfiguration(preferenceKey);
				}
			});
			return true;
		}
	});
}
```

and then just add the calls:

```
addPreferenceHandler(ZeTimeConstants.PREF_TIME_FORMAT);
addPreferenceHandler(ZeTimeConstants.PREF_ANTI_LOSS_SIGNALING);
addPreferenceHandler(ZeTimeConstants.PREF_SMS_SIGNALING);
```
etc. Saves probably 80% of the lines in the class.

Reply:
Sure, I think I get your idea. I will try it out, saving some lines is always a good choice. ;)
This my first project in android app and java development, so I'm still learning by looking at the code of other devices here.

Reply:
No offense intended, you do a great job! We can probably make use of this method elsewhere too, maybe with an additional argument for the string for onSendConfiguration().